packages/dds/tree/src/simple-tree/simpleSchema.ts

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Kind of {@link TreeNodeSchema}.
+ * @alpha
+ */
+export type SimpleNodeSchemaKind = "object" | "array" | "map" | "leaf";
+
+/**
+ * Kind of {@link FieldSchema}.
+ */
+export type SimpleFieldSchemaKind = "optional" | "required" | "identifier";
+
+/**
+ * Kind of leaf schema.
+ */
+export type SimpleLeafSchemaKind = "string" | "number" | "boolean" | "null" | "fluid-handle";
+
+/**
+ * Base interface for all {@link SimpleNodeSchema} implementations.
+ */
+export interface SimpleNodeSchemaBase<TNodeKind extends SimpleNodeSchemaKind> {
+	/**
+	 * The kind of {@link SimpleNodeSchema}.
+	 *
+	 * @remarks can be used to type-switch between implementations.
+	 */
+	readonly kind: TNodeKind;
+}
+
+/**
+ * A {@link SimpleNodeSchema} for an object node.
+ */
+export interface SimpleObjectNodeSchema extends SimpleNodeSchemaBase<"object"> {
+	/**
+	 * Schemas for each of the object's fields, keyed off of schema's view keys.
+	 */
+	readonly fields: Record<string, SimpleFieldSchema>;
+}
+
+/**
+ * A {@link SimpleNodeSchema} for an array node.
+ */
+export interface SimpleArrayNodeSchema extends SimpleNodeSchemaBase<"array"> {
+	/**
+	 * The types allowed in the array.
+	 *
+	 * @remarks Refers to the types by identifier.
+	 * A {@link SimpleTreeSchema} is needed to resolve these identifiers to their schema {@link SimpleTreeSchema.definitions}.
+	 */
+	readonly allowedTypes: ReadonlySet<string>;
+}
+
+/**
+ * A {@link SimpleNodeSchema} for a map node.
+ */
+export interface SimpleMapNodeSchema extends SimpleNodeSchemaBase<"map"> {
+	/**
+	 * The types allowed as values in the map.
+	 *
+	 * @remarks Refers to the types by identifier.
+	 * A {@link SimpleTreeSchema} is needed to resolve these identifiers to their schema {@link SimpleTreeSchema.definitions}.
+	 */
+	readonly allowedTypes: ReadonlySet<string>;
+}
+
+/**
+ * A {@link SimpleNodeSchema} for a leaf node.
+ */
+export interface SimpleLeafNodeSchema extends SimpleNodeSchemaBase<"leaf"> {
+	/**
+	 * The type of leaf node.
+	 */
+	readonly type: SimpleLeafSchemaKind;
```

Comment:
Perhaps "leafType" rather than just "type"? "type" is often the property name used as the union discriminator, but we already have "kind" on the base type that does that. And it's muddled further in that the _type_ of "type" is "...Kind", but the "type" is not the _kind_, lol :D 

If you like just "type" (I'm okay with it), at least rename "SimpleLeafSchemaKind" to "SimpleLeafSchemaType".

Reply:
Good call. I'll rename this to "leafKind" to be explicit and consistent.